notes/legend-connector-refactoring-notes.md

```diff
@@ -0,0 +1,115 @@
+            Properties connectionProperties = new Properties();
+            connectionProperties.putAll(properties);
+            connectionProperties.put("user", "fred");
+            connectionProperties.put("password", userPasswordAuthenticationStrategy.getPassword());
+            return Tuples.pair(url, connectionProperties);
+        } 
+        else if ( ...)
+        {
+            // do something else
+        }
+        else if ( ...)
+        {
+          // do something else
+        }        
+        return Tuples.pair(url, properties);
+    }
+```
+### New module per authentication strategy 
+
+E.g legend-engine-executionPlan-execution-store-relational-connection-authn-userpass
+
+### Delete RelationalDatabaseCommandsVisitor 
+
+The StreamResultToTempTableVisitor was causing dependency cycles.
+
+Temp table handling etc is now pushed into RelationalDatabaseCommands. 
+
+### Delete DataSourceIdentifiersCaseSensitiveVisitor 
+
+Same idea as temp table handling. Push database specific code into the plugin and related classes.
+
+```
+    // RelationalExecutor 
+    
+    private void buildTransformersAndBuilder(ExecutionNode node, DatabaseConnection databaseConnection) throws SQLException
+    {
+        // TODO : epsstan : ask the data source about this ??
+        RelationalDatabaseConnection relationalDatabaseConnection = (RelationalDatabaseConnection) databaseConnection;
+        RelationalConnectionPlugin relationalConnectionPlugin = new RelationalConnectionPluginLoader().getPlugin(relationalDatabaseConnection.databaseType);
+        DataSourceSpecification dataSourceSpecification = relationalConnectionPlugin.buildDatasourceSpecification(relationalDatabaseConnection, new RelationalExecutorInfo());
+        boolean isDatabaseIdentifiersCaseSensitive = dataSourceSpecification.isDatabaseIdentifiersCaseSensitive();
+       ...
+    }
+```
+
+Questions / Gotchas
+==================
+### Can we remove the datasource specification key classes ??
+
+We use the datasource specification key to index user + database specific connections.
+
+Can we introduce a method that returns a key in the protocol classes and remove the key classes ??
+
+### Can we remove the authentication strategy key classes ??
+
+Similarly can we remove the auth strategy key classes
+
+### Overspecified/Long keys
+
+In the example below, "quoteIdentifiers" does not uniquely identify the database
+
+```
+    @Override
+    public String shortId()
+    {
+        return "Snowflake_" +
+                "account:" + accountName + "_" +
+                "region:" + region + "_" +
+                "warehouse:" + warehouseName + "_" +
+                "db:" + databaseName + "_" +
+                "cloudType:" + cloudType + "_" +
+                "quoteIdentifiers:" + quoteIdentifiers;
+    }
+```
+
+### OAuth profiles
+TODO 
+
+### Plugin load order
+TODO 
```

Comment:
Plugins should be independent of each other.
Can you please remind me why the order matters